libpod/container_internal.go

```diff
@@ -2738,3 +2741,327 @@ func (c *Container) update(resources *spec.LinuxResources, restartPolicy *string
 
 	return nil
 }
+
+func (c *Container) getCopyOfHealthCheckAndStartupHelathCheck() (*manifest.Schema2HealthConfig, *define.StartupHealthCheck) {
+	var healthCheck manifest.Schema2HealthConfig
+	if c.config.HealthCheckConfig != nil {
+		healthCheck = manifest.Schema2HealthConfig{
+			Test:          slices.Clone(c.config.HealthCheckConfig.Test),
+			Interval:      c.config.HealthCheckConfig.Interval,
+			Retries:       c.config.HealthCheckConfig.Retries,
+			StartInterval: c.config.HealthCheckConfig.StartInterval,
+			StartPeriod:   c.config.HealthCheckConfig.StartPeriod,
+			Timeout:       c.config.HealthCheckConfig.Timeout,
+		}
+	}
+
+	var startupHealthCheck define.StartupHealthCheck
+	if c.config.StartupHealthCheckConfig != nil {
+		startupHealthCheck = define.StartupHealthCheck{
+			Successes: c.config.StartupHealthCheckConfig.Successes,
+		}
+		startupHealthCheck.Test = slices.Clone(c.config.StartupHealthCheckConfig.Test)
+		startupHealthCheck.Interval = c.config.StartupHealthCheckConfig.Interval
+		startupHealthCheck.Retries = c.config.StartupHealthCheckConfig.Retries
+		startupHealthCheck.StartInterval = c.config.StartupHealthCheckConfig.StartInterval
+		startupHealthCheck.StartPeriod = c.config.StartupHealthCheckConfig.StartPeriod
+		startupHealthCheck.Timeout = c.config.StartupHealthCheckConfig.Timeout
+	}
+	return &healthCheck, &startupHealthCheck
+}
+
+func (c *Container) changeHealthCheckConfiguration(changedHealthCheckConfig *entities.UpdateHealthCheckConfig) (bool, error) {
```

Comment:
I don't like leaking entities into Libpod. Maybe move UpdateHealthCheckConfig into libpod and make the entities version just contain it?

Reply:
Actually, no. We should probably move this parsing code out of Libpod. Instead we should accept a `manifest.Schema2HealthConfig` and `define.StartupHealthCheck` in Update in libpod, and do all the validation in the frontend.